projects/dsp-ui/src/lib/viewer/representation/upload-file/upload-file.component.ts

```diff
@@ -100,29 +110,45 @@ export class UploadFileComponent implements OnInit {
     }
 
     /**
-     * Converts date to readable format
+     * Converts date to a readable format.
      * @param date date to be converted
      */
     convertDate(date: number): string {
         return new Date(+`Date(${date})`.replace(/\D/g, '')).toLocaleDateString();
     }
 
     /**
-     * Removes the attachement
+     * Removes the attachment
      */
     deleteAttachment(): void {
-        this.file = null;
-        this.thumbnaillUrl = null;
         this.fileControl.reset();
     }
 
     /**
      * Initializes form group
      */
     initializeForm(): void {
+        this.fileControl = new FormControl(null, Validators.required);
+
+        this.fileControl.valueChanges.subscribe(
+            val => {
+                // check if the form has been reset
+                if (val === null) {
+                    this.file = null;
+                    this.thumbnailUrl = null;
+                }
+            }
+        );
+
         this.form = this._fb.group({
-            file: [null, Validators.required]
+            file: this.fileControl
         }, { updateOn: 'blur' });
+
+        if (this.parentForm !== undefined) {
```

Comment:
Is this in purpose just for undefined? Otherwise it can be simplified to `if (this.parentForm) {}`

Reply:
Of course `if (this.parentForm) {}` would also work.

However, this involves a [conversion of `undefined` to a Boolean value](https://www.w3schools.com/js/js_type_conversion.asp).

`this.parentForm !== undefined` checks that the member `this.parentForm` has been initialised, [not performing any type conversion](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_inequality). 

So all it does is checking exactly for `undefined` which is a special type in JS. 

`if (myvar) {}` could give unexpected results if `myvar` was actually a Boolean. So if `mayvar` was initialised to `false`, this would not be the same as `undefined`.

The condition should prevent errors in case of non initialised vars.

So in general, I am trying to avoid implicit type conversions.